lib/catalog/owid/catalog/tables.py

```diff
@@ -963,21 +962,48 @@ def _add_table_and_variables_metadata_to_table(table: Table, metadata: Optional[
 
 
 def read_csv(
-    filepath_or_buffer: Union[FilePath, ReadCsvBuffer[bytes], ReadCsvBuffer[str]],
+    path: Union[str, Path],
```

Comment:
What is the motivation for this argument name change? I would stick to the old name, which is the one that `pandas` uses.

Reply:
I just thought that `filepath_or_buffer` is a long, hard to understand argument, while `path` is much clearer (and we will pass a path in the vast majority of cases). Function `read_excel` has as first argument `io` which is also ambiguous. That's why I used `path` on both. Also, we are already introducing `metadata` and `underscore`, so the list of arguments is already different to `pandas`. Do you see any harm in changing these argument names?

Reply:
In general, I'd minimise argument names whenever they are precisely equivalent. It is easier to get documentation on them (directly from pandas). Also, maintenance might be easier (as in if pandas were to introduce some API changes, argument descriptions, etc.).

I would not change `filepath_or_buffer` to `path`.

If it were the case that we _only_ accept paths (and not buffers), I'd make that explicit in the docs and adapt the code accordingly to raise an error if a buffer is passed.


Adding new ones, I think, is fine. And users should only use our docs for that.

Reply:
OK, thanks, I'll keep this in mind on the next refactor when handling metadata (coming soon). The first mandatory argument of these functions is usually not explicitly called, so the name is not very important. Also, keep in mind that:
* We want to eventually create our own functions, instead of inheriting pandas behaviour. So it's likely that we'll end up changing names.
* I found it a bit risky to import obscure pandas types like `ReadCsvBuffer` from pandas hidden modules (see line `from pandas._typing import FilePath, ReadCsvBuffer  # type: ignore`). I think it would be safer to define those types ourselves.

Reply:
I still think that, even if we are re-defining names and customising the functions, we should try to avoid distancing ourselves too much from pandas whenever possible, as most people are used to it.

If we are only accepting paths, we can indeed change the name and have a check like `if not isinstance(path, str): raise TypeError("...")`.

Reply:
I just created https://github.com/owid/etl/pull/1327